app/lib/features/super_invites/providers/super_invites_providers.dart

```diff
@@ -31,8 +31,30 @@ final superInvitesForRoom = FutureProvider.autoDispose
       .toList();
 });
 
+/// Get SuperInviteToken that is associate with single roomId only
+final inviteCodeForSelectRoomOnly = FutureProvider.autoDispose
+    .family<SuperInviteToken?, String>((ref, roomId) async {
+  // Get all tokens
+  final allInvites = await ref.watch(superInvitesTokensProvider.future);
+  // Get list of tokens with associate with given roomId
+  var allInvitesRelatedToRoomId = allInvites
+      .where(
+        (invite) =>
+            invite.rooms().map((e) => e.toDartString()).contains(roomId),
+      )
+      .toList();
```

Comment:
this should reuse the existing provider rather than copy-paste the code ...
```suggestion
  final allInvitesRelatedToRoomId = await ref.watch(superInvietsForRoom(roomId).future);
```

Reply:
Created additional provide to manage multiple invite code for specific room more precisely. Existing provide will return multiple tokens if available and on the other side, new provider which I created will return only single token or null for specific roomId. I will explain this on the call if it is confusing.

Reply:
no, that's fine, if you look at my diff you'll see that I only asked for stuff to be reused...

Reply:
Ah, understood. Noted!